textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/LuceneDataReader.java

```diff
@@ -44,14 +40,11 @@
     private IndexSearcher indexSearcher;
     private ScoreDoc[] scoreDocs;
     private IndexReader indexReader;
-    private String query;
-    private String defaultField;
+    private Query query;
     
-    public LuceneDataReader(IDataStore dataStore, 
-            String query, String defaultField) {
+    public LuceneDataReader(IDataStore dataStore, Query query) {

```

Comment:
Here's my understanding.  The analyzer for a writer is needed to analyze/tokenize documents before writing them to the disk.  The analyzer in the query of a reader is needed to analyzer/tokenize the query string.

If you agree with this understanding, then we are on the same page.  The rest of this PR looks good to me.